### ISSUE TITLE
Replace --model-options with Clang method for backend/frontend args

### ISSUE THREAD

**Issue by jjwilke** [sstsimulator/sst-core#360]
We current have to use --model-options to pass through command line args to our underlying python script. Another possibility, used in projects like Clang, is to use the command line token '--' to 'split' the list.

Instead of
````
sst myPython.py --model-options="here are args"
````

you would use

````
sst myPython.py -- here are args
````

It doesn't seem like a dramatic change, but makes it much easier to specify command line flags for the python script

**Comment by feldergast** [comment]
This worked at one point.  Maybe when we ripped the boost ProgramOptions out we lost this.  I'll look into it.

**Comment by nmhamster** [comment]
@feldergast / @jjwilke - I don't think this did ever work. Brian Barrett veto'd this when we were designing this solution saying that it had been a major mistake in some MPI implementations. On his recommendation we avoiding implementing it as far as I can remember. That's why we went back to model options.

**Comment by feldergast** [comment]
I implemented it after Brian left.  At this point, I don't remember for sure if it was only in a branch or if I pushed it into devel.  In either case, I think it makes sense to implement now if there are no objections.

**Comment by nmhamster** [comment]
Well I still want to understand why Brian didn't want to do it before we proceed. He had good insight into software design and the pitfalls. Why is this any better or worse than just using the current `--model-options` command line?

**Comment by jjwilke** [comment]
It just gets a little bit difficult if model-options is itself command line options to another tool (which is the case in macro). It's a mess of nested quotes (and sometimes would require triply-nested). SST/macro was using an app-argv (which we might replace with '--').

````
--model-options="--exe=lulesh --app-argv='8 8 8'"
````

versus
````
-- --exe=lulesh -- 8 8 8
````

Another option would be namespacing:
````
--model:exe=lulesh --model:app-argv="8 8 8"
````

The current setup is okay, but cumbersome in a few cases. The use of "--" seems to occur several places (args to the tool in front, args to pass through in back) - so it was just a possibility to consider as "standard practice" more than a "this is terrible and we can't use it anymore."

**Comment by feldergast** [comment]
This is actually already in the code, just not documented.  Added info in the --help output to discuss this capability.

**Comment by feldergast** [comment]
This was addressed in #590
